Import pandas

In [1]:
import pandas as pd
pd.set_option('display.max_rows', 200)

Carregar o dataframe `escolas` do arquivo csv.

In [2]:
escolas = pd.read_csv('data/escolas.csv')
escolas

id                                     Escolas_Postos  \
0    178        CENTRO INTEGRADO DE EDUCAÇÃO PÚBLICA HENFIL   
1    634                         EM ALICE DO AMARAL PEIXOTO   
2    483                                 EM CELESTINO SILVA   
3    476                  ESCOLA MUNICIPAL FLORIANO PEIXOTO   
4    132                                  EM PEREIRA PASSOS   
5     17                       E.M PRESIDENTE JOSE LINHARES   
6     55                                  EM SANTA CATARINA   
7    490                                      E.M TIA CIATA   
8    600                                         EM URUGUAI   
9     89                                   EM MARIO CLAUDIO   
10   387                     CIEP PRESIDENTE AGOSTINHO NETO   
11   331                                CIEP TANCREDO NEVES   
12   450                                   EM ALBERTO BARTH   
13   558                                     EM CÍCERO PENA   
14   413                                  EM CÓCIO BARCELOS   
15   413                                  EM CÓCIO BARCELOS   
16   540                              EM HENRIQUE DODSWORTH   
17   309                                          EM MÉXICO   
18   268                      ESCOLA MUNICIPAL MINAS GERAIS   
19   523                     EM PRESIDENTE DA COSTA E SILVA   
20    17                       E.M PRESIDENTE JOSE LINHARES   
21    35                                  EM SENADOR CORREA   
22    11                                 CIEP SAMUEL WEINER   
23   226                                 EM AFRANIO PEIXOTO   
24   638                                  E.M AZEVEDO SODRÉ   
25    89                                   EM MARIO CLAUDIO   
26   126                                 EM DUQUE DE CAXIAS   
27   180                           ESCOLA MUNICIPAL EQUADOR   
28   163                                 EM LAUDIMIA TROTTA   
29   361                              EM PRUDENTE DE MORAES   
30   280   ESCOLA MUNICIPAL GENERAL HUMBERTO DE SOUZA MELLO   
31    60               ESCOLA MUNICIPAL BARÃO HOMEM DE MELO   
32   587                             EM ALMIRANTE TAMANDARÉ   
33   653                            EM LÚCIA MIGUEL PEREIRA   
34   660                       E.M NOSSA SENHORA DA POMPÉIA   
35   382                                    EM SANTOS ANJOS   
36   217                                         EM ALAGOAS   
37   378                           EM BRIGADEIRO FARIA LIMA   
38   429                                   EM FELIX PACHECO   
39    25                     ESCOLA MUNICIPAL GEORGE SUMMER   
40   660                       E.M NOSSA SENHORA DA POMPÉIA   
41   373                       EM PROFESSOR AUGUSTO PAULINO   
42   624                               EM RIO GRANDE DO SUL   
43   157                                     E.M. SARMIENTO   
44    20                                          EM TAGORE   
45   117                        CIEP CHANCELER WILLY BRANDT   
46   502                               CIEP PATRICE LUMUMBA   
47   351  CENTRO INTEGRADO DE EDUCAÇÃO PÚBLICA VINÍCIUS ...   
48   135                                EM DOMINGOS BEBIANO   
49   388                                  EM CARDEAL CAMARA   
50   440                                 EM JOAQUIM RIBEIRO   
51   259                                      EM PERNAMBUCO   
52   422                 ESCOLA MUNICIPAL ALBINO SOUZA CRUZ   
53    41                                   EM ANDRADE NEVES   
54   537                                   EM ARY QUINTELLA   
55   230                                          EM BERLIM   
56   379                                CIEP NELSON MANDELA   
57    89                                   EM MARIO CLAUDIO   
58   146                            EM BERNARDO VASCONCELOS   
59   566                                     EM BRANT HORTA   
60   388                                  EM CARDEAL CAMARA   
61   498                                           EM CHILE   
62   663                          EM 

Renomear as colunas de `escolas` e aplicar o padrão snake case.

In [3]:
column_mapping = {
    'id': 'id',
    'Escolas_Postos': 'nome_escola',
    'BAIRRO': 'bairro',
    'ENDEREÇO ': 'logradouro_entrega',
    'lat': 'latitude',
    'lon': 'longitude'
}

escolas.rename(columns=column_mapping,inplace=True)
escolas

id                                     escolas_postos  \
0    178        CENTRO INTEGRADO DE EDUCAÇÃO PÚBLICA HENFIL   
1    634                         EM ALICE DO AMARAL PEIXOTO   
2    483                                 EM CELESTINO SILVA   
3    476                  ESCOLA MUNICIPAL FLORIANO PEIXOTO   
4    132                                  EM PEREIRA PASSOS   
5     17                       E.M PRESIDENTE JOSE LINHARES   
6     55                                  EM SANTA CATARINA   
7    490                                      E.M TIA CIATA   
8    600                                         EM URUGUAI   
9     89                                   EM MARIO CLAUDIO   
10   387                     CIEP PRESIDENTE AGOSTINHO NETO   
11   331                                CIEP TANCREDO NEVES   
12   450                                   EM ALBERTO BARTH   
13   558                                     EM CÍCERO PENA   
14   413                                  EM CÓCIO BARCELOS   
15   413                                  EM CÓCIO BARCELOS   
16   540                              EM HENRIQUE DODSWORTH   
17   309                                          EM MÉXICO   
18   268                      ESCOLA MUNICIPAL MINAS GERAIS   
19   523                     EM PRESIDENTE DA COSTA E SILVA   
20    17                       E.M PRESIDENTE JOSE LINHARES   
21    35                                  EM SENADOR CORREA   
22    11                                 CIEP SAMUEL WEINER   
23   226                                 EM AFRANIO PEIXOTO   
24   638                                  E.M AZEVEDO SODRÉ   
25    89                                   EM MARIO CLAUDIO   
26   126                                 EM DUQUE DE CAXIAS   
27   180                           ESCOLA MUNICIPAL EQUADOR   
28   163                                 EM LAUDIMIA TROTTA   
29   361                              EM PRUDENTE DE MORAES   
30   280   ESCOLA MUNICIPAL GENERAL HUMBERTO DE SOUZA MELLO   
31    60               ESCOLA MUNICIPAL BARÃO HOMEM DE MELO   
32   587                             EM ALMIRANTE TAMANDARÉ   
33   653                            EM LÚCIA MIGUEL PEREIRA   
34   660                       E.M NOSSA SENHORA DA POMPÉIA   
35   382                                    EM SANTOS ANJOS   
36   217                                         EM ALAGOAS   
37   378                           EM BRIGADEIRO FARIA LIMA   
38   429                                   EM FELIX PACHECO   
39    25                     ESCOLA MUNICIPAL GEORGE SUMMER   
40   660                       E.M NOSSA SENHORA DA POMPÉIA   
41   373                       EM PROFESSOR AUGUSTO PAULINO   
42   624                               EM RIO GRANDE DO SUL   
43   157                                     E.M. SARMIENTO   
44    20                                          EM TAGORE   
45   117                        CIEP CHANCELER WILLY BRANDT   
46   502                               CIEP PATRICE LUMUMBA   
47   351  CENTRO INTEGRADO DE EDUCAÇÃO PÚBLICA VINÍCIUS ...   
48   135                                EM DOMINGOS BEBIANO   
49   388                                  EM CARDEAL CAMARA   
50   440                                 EM JOAQUIM RIBEIRO   
51   259                                      EM PERNAMBUCO   
52   422                 ESCOLA MUNICIPAL ALBINO SOUZA CRUZ   
53    41                                   EM ANDRADE NEVES   
54   537                                   EM ARY QUINTELLA   
55   230                                          EM BERLIM   
56   379                                CIEP NELSON MANDELA   
57    89                                   EM MARIO CLAUDIO   
58   146                            EM BERNARDO VASCONCELOS   
59   566                                     EM BRANT HORTA   
60   388                                  EM CARDEAL CAMARA   
61   498                                           EM CHILE   
62   663                          EM 

Mudar o tipo de dados da coluna `id`para string e preencher com zeros (à esquerda) até len = 3.

In [4]:
escolas['id'] = escolas['id'].astype(str).str.zfill(3)
escolas

id                                     escolas_postos  \
0    178        CENTRO INTEGRADO DE EDUCAÇÃO PÚBLICA HENFIL   
1    634                         EM ALICE DO AMARAL PEIXOTO   
2    483                                 EM CELESTINO SILVA   
3    476                  ESCOLA MUNICIPAL FLORIANO PEIXOTO   
4    132                                  EM PEREIRA PASSOS   
5    017                       E.M PRESIDENTE JOSE LINHARES   
6    055                                  EM SANTA CATARINA   
7    490                                      E.M TIA CIATA   
8    600                                         EM URUGUAI   
9    089                                   EM MARIO CLAUDIO   
10   387                     CIEP PRESIDENTE AGOSTINHO NETO   
11   331                                CIEP TANCREDO NEVES   
12   450                                   EM ALBERTO BARTH   
13   558                                     EM CÍCERO PENA   
14   413                                  EM CÓCIO BARCELOS   
15   413                                  EM CÓCIO BARCELOS   
16   540                              EM HENRIQUE DODSWORTH   
17   309                                          EM MÉXICO   
18   268                      ESCOLA MUNICIPAL MINAS GERAIS   
19   523                     EM PRESIDENTE DA COSTA E SILVA   
20   017                       E.M PRESIDENTE JOSE LINHARES   
21   035                                  EM SENADOR CORREA   
22   011                                 CIEP SAMUEL WEINER   
23   226                                 EM AFRANIO PEIXOTO   
24   638                                  E.M AZEVEDO SODRÉ   
25   089                                   EM MARIO CLAUDIO   
26   126                                 EM DUQUE DE CAXIAS   
27   180                           ESCOLA MUNICIPAL EQUADOR   
28   163                                 EM LAUDIMIA TROTTA   
29   361                              EM PRUDENTE DE MORAES   
30   280   ESCOLA MUNICIPAL GENERAL HUMBERTO DE SOUZA MELLO   
31   060               ESCOLA MUNICIPAL BARÃO HOMEM DE MELO   
32   587                             EM ALMIRANTE TAMANDARÉ   
33   653                            EM LÚCIA MIGUEL PEREIRA   
34   660                       E.M NOSSA SENHORA DA POMPÉIA   
35   382                                    EM SANTOS ANJOS   
36   217                                         EM ALAGOAS   
37   378                           EM BRIGADEIRO FARIA LIMA   
38   429                                   EM FELIX PACHECO   
39   025                     ESCOLA MUNICIPAL GEORGE SUMMER   
40   660                       E.M NOSSA SENHORA DA POMPÉIA   
41   373                       EM PROFESSOR AUGUSTO PAULINO   
42   624                               EM RIO GRANDE DO SUL   
43   157                                     E.M. SARMIENTO   
44   020                                          EM TAGORE   
45   117                        CIEP CHANCELER WILLY BRANDT   
46   502                               CIEP PATRICE LUMUMBA   
47   351  CENTRO INTEGRADO DE EDUCAÇÃO PÚBLICA VINÍCIUS ...   
48   135                                EM DOMINGOS BEBIANO   
49   388                                  EM CARDEAL CAMARA   
50   440                                 EM JOAQUIM RIBEIRO   
51   259                                      EM PERNAMBUCO   
52   422                 ESCOLA MUNICIPAL ALBINO SOUZA CRUZ   
53   041                                   EM ANDRADE NEVES   
54   537                                   EM ARY QUINTELLA   
55   230                                          EM BERLIM   
56   379                                CIEP NELSON MANDELA   
57   089                                   EM MARIO CLAUDIO   
58   146                            EM BERNARDO VASCONCELOS   
59   566                                     EM BRANT HORTA   
60   388                                  EM CARDEAL CAMARA   
61   498                                           EM CHILE   
62   663                          EM 

Substituir os valores da coluna `bairro` em `escolas` com os valores corretos do dataframe `subprefeituras`.
Para alguns valores, como 'Maracanã/ Vila Isabel', 'Maracanã/ Tijuca', 'Recreio', 'Rio Das Pedras', 'Freguesia Jpa', 'Augusto Vasconcelos' e 'Nova Sepetiba', foi feita busca na internet para identificar o bairro correto.

In [5]:
bairros = {
'HUMAITA':'HUMAITÁ', 
'MARACANÃ/ VILA ISABEL':'VILA ISABEL',
'MARACANÃ/ TIJUCA':'VILA ISABEL',
'MEIER':'MÉIER', 
'INHAUMA':'INHAÚMA',
'OSWALDO CRUZ':'OSVALDO CRUZ',
'GARDENIA AZUL':'GARDÊNIA AZUL',
'RECREIO':'RECREIO DOS BANDEIRANTES',
'RIO DAS PEDRAS':'JACAREPAGUÁ',
'FREGUESIA JPA': 'FREGUESIA (JACAREPAGUÁ)',
'AUGUSTO VASCONCELOS':'SENADOR VASCONCELOS',
'PACIENCIA':'PACIÊNCIA',
'NOVA SEPETIBA': 'SEPETIBA'
}

escolas1 = escolas
escolas1['bairro'] = escolas1['bairro'].replace(bairros)
escolas1

id                                     escolas_postos  \
0    178        CENTRO INTEGRADO DE EDUCAÇÃO PÚBLICA HENFIL   
1    634                         EM ALICE DO AMARAL PEIXOTO   
2    483                                 EM CELESTINO SILVA   
3    476                  ESCOLA MUNICIPAL FLORIANO PEIXOTO   
4    132                                  EM PEREIRA PASSOS   
5    017                       E.M PRESIDENTE JOSE LINHARES   
6    055                                  EM SANTA CATARINA   
7    490                                      E.M TIA CIATA   
8    600                                         EM URUGUAI   
9    089                                   EM MARIO CLAUDIO   
10   387                     CIEP PRESIDENTE AGOSTINHO NETO   
11   331                                CIEP TANCREDO NEVES   
12   450                                   EM ALBERTO BARTH   
13   558                                     EM CÍCERO PENA   
14   413                                  EM CÓCIO BARCELOS   
15   413                                  EM CÓCIO BARCELOS   
16   540                              EM HENRIQUE DODSWORTH   
17   309                                          EM MÉXICO   
18   268                      ESCOLA MUNICIPAL MINAS GERAIS   
19   523                     EM PRESIDENTE DA COSTA E SILVA   
20   017                       E.M PRESIDENTE JOSE LINHARES   
21   035                                  EM SENADOR CORREA   
22   011                                 CIEP SAMUEL WEINER   
23   226                                 EM AFRANIO PEIXOTO   
24   638                                  E.M AZEVEDO SODRÉ   
25   089                                   EM MARIO CLAUDIO   
26   126                                 EM DUQUE DE CAXIAS   
27   180                           ESCOLA MUNICIPAL EQUADOR   
28   163                                 EM LAUDIMIA TROTTA   
29   361                              EM PRUDENTE DE MORAES   
30   280   ESCOLA MUNICIPAL GENERAL HUMBERTO DE SOUZA MELLO   
31   060               ESCOLA MUNICIPAL BARÃO HOMEM DE MELO   
32   587                             EM ALMIRANTE TAMANDARÉ   
33   653                            EM LÚCIA MIGUEL PEREIRA   
34   660                       E.M NOSSA SENHORA DA POMPÉIA   
35   382                                    EM SANTOS ANJOS   
36   217                                         EM ALAGOAS   
37   378                           EM BRIGADEIRO FARIA LIMA   
38   429                                   EM FELIX PACHECO   
39   025                     ESCOLA MUNICIPAL GEORGE SUMMER   
40   660                       E.M NOSSA SENHORA DA POMPÉIA   
41   373                       EM PROFESSOR AUGUSTO PAULINO   
42   624                               EM RIO GRANDE DO SUL   
43   157                                     E.M. SARMIENTO   
44   020                                          EM TAGORE   
45   117                        CIEP CHANCELER WILLY BRANDT   
46   502                               CIEP PATRICE LUMUMBA   
47   351  CENTRO INTEGRADO DE EDUCAÇÃO PÚBLICA VINÍCIUS ...   
48   135                                EM DOMINGOS BEBIANO   
49   388                                  EM CARDEAL CAMARA   
50   440                                 EM JOAQUIM RIBEIRO   
51   259                                      EM PERNAMBUCO   
52   422                 ESCOLA MUNICIPAL ALBINO SOUZA CRUZ   
53   041                                   EM ANDRADE NEVES   
54   537                                   EM ARY QUINTELLA   
55   230                                          EM BERLIM   
56   379                                CIEP NELSON MANDELA   
57   089                                   EM MARIO CLAUDIO   
58   146                            EM BERNARDO VASCONCELOS   
59   566                                     EM BRANT HORTA   
60   388                                  EM CARDEAL CAMARA   
61   498                                           EM CHILE   
62   663                          EM 

Transformar todas as colunas com tipo de dados string em `escolas` para upper case.

In [6]:
escolas2 = escolas1.apply(lambda x:x.astype(str).str.upper())
escolas2

id                                     escolas_postos  \
0    178        CENTRO INTEGRADO DE EDUCAÇÃO PÚBLICA HENFIL   
1    634                         EM ALICE DO AMARAL PEIXOTO   
2    483                                 EM CELESTINO SILVA   
3    476                  ESCOLA MUNICIPAL FLORIANO PEIXOTO   
4    132                                  EM PEREIRA PASSOS   
5    017                       E.M PRESIDENTE JOSE LINHARES   
6    055                                  EM SANTA CATARINA   
7    490                                      E.M TIA CIATA   
8    600                                         EM URUGUAI   
9    089                                   EM MARIO CLAUDIO   
10   387                     CIEP PRESIDENTE AGOSTINHO NETO   
11   331                                CIEP TANCREDO NEVES   
12   450                                   EM ALBERTO BARTH   
13   558                                     EM CÍCERO PENA   
14   413                                  EM CÓCIO BARCELOS   
15   413                                  EM CÓCIO BARCELOS   
16   540                              EM HENRIQUE DODSWORTH   
17   309                                          EM MÉXICO   
18   268                      ESCOLA MUNICIPAL MINAS GERAIS   
19   523                     EM PRESIDENTE DA COSTA E SILVA   
20   017                       E.M PRESIDENTE JOSE LINHARES   
21   035                                  EM SENADOR CORREA   
22   011                                 CIEP SAMUEL WEINER   
23   226                                 EM AFRANIO PEIXOTO   
24   638                                  E.M AZEVEDO SODRÉ   
25   089                                   EM MARIO CLAUDIO   
26   126                                 EM DUQUE DE CAXIAS   
27   180                           ESCOLA MUNICIPAL EQUADOR   
28   163                                 EM LAUDIMIA TROTTA   
29   361                              EM PRUDENTE DE MORAES   
30   280   ESCOLA MUNICIPAL GENERAL HUMBERTO DE SOUZA MELLO   
31   060               ESCOLA MUNICIPAL BARÃO HOMEM DE MELO   
32   587                             EM ALMIRANTE TAMANDARÉ   
33   653                            EM LÚCIA MIGUEL PEREIRA   
34   660                       E.M NOSSA SENHORA DA POMPÉIA   
35   382                                    EM SANTOS ANJOS   
36   217                                         EM ALAGOAS   
37   378                           EM BRIGADEIRO FARIA LIMA   
38   429                                   EM FELIX PACHECO   
39   025                     ESCOLA MUNICIPAL GEORGE SUMMER   
40   660                       E.M NOSSA SENHORA DA POMPÉIA   
41   373                       EM PROFESSOR AUGUSTO PAULINO   
42   624                               EM RIO GRANDE DO SUL   
43   157                                     E.M. SARMIENTO   
44   020                                          EM TAGORE   
45   117                        CIEP CHANCELER WILLY BRANDT   
46   502                               CIEP PATRICE LUMUMBA   
47   351  CENTRO INTEGRADO DE EDUCAÇÃO PÚBLICA VINÍCIUS ...   
48   135                                EM DOMINGOS BEBIANO   
49   388                                  EM CARDEAL CAMARA   
50   440                                 EM JOAQUIM RIBEIRO   
51   259                                      EM PERNAMBUCO   
52   422                 ESCOLA MUNICIPAL ALBINO SOUZA CRUZ   
53   041                                   EM ANDRADE NEVES   
54   537                                   EM ARY QUINTELLA   
55   230                                          EM BERLIM   
56   379                                CIEP NELSON MANDELA   
57   089                                   EM MARIO CLAUDIO   
58   146                            EM BERNARDO VASCONCELOS   
59   566                                     EM BRANT HORTA   
60   388                                  EM CARDEAL CAMARA   
61   498                                           EM CHILE   
62   663                          EM 

Remover acentos de todas as colunas com tipo de dados string em `escolas`.

In [7]:
cols = escolas2.select_dtypes(include=[object]).columns
escolas3 = escolas2
escolas3[cols] = escolas2.apply(lambda x:x.str.normalize('NFKD').str.encode('ascii',errors='ignore').str.decode('utf-8'))
escolas3

id                                     escolas_postos  \
0    178        CENTRO INTEGRADO DE EDUCACAO PUBLICA HENFIL   
1    634                         EM ALICE DO AMARAL PEIXOTO   
2    483                                 EM CELESTINO SILVA   
3    476                  ESCOLA MUNICIPAL FLORIANO PEIXOTO   
4    132                                  EM PEREIRA PASSOS   
5    017                       E.M PRESIDENTE JOSE LINHARES   
6    055                                  EM SANTA CATARINA   
7    490                                      E.M TIA CIATA   
8    600                                         EM URUGUAI   
9    089                                   EM MARIO CLAUDIO   
10   387                     CIEP PRESIDENTE AGOSTINHO NETO   
11   331                                CIEP TANCREDO NEVES   
12   450                                   EM ALBERTO BARTH   
13   558                                     EM CICERO PENA   
14   413                                  EM COCIO BARCELOS   
15   413                                  EM COCIO BARCELOS   
16   540                              EM HENRIQUE DODSWORTH   
17   309                                          EM MEXICO   
18   268                      ESCOLA MUNICIPAL MINAS GERAIS   
19   523                     EM PRESIDENTE DA COSTA E SILVA   
20   017                       E.M PRESIDENTE JOSE LINHARES   
21   035                                  EM SENADOR CORREA   
22   011                                 CIEP SAMUEL WEINER   
23   226                                 EM AFRANIO PEIXOTO   
24   638                                  E.M AZEVEDO SODRE   
25   089                                   EM MARIO CLAUDIO   
26   126                                 EM DUQUE DE CAXIAS   
27   180                           ESCOLA MUNICIPAL EQUADOR   
28   163                                 EM LAUDIMIA TROTTA   
29   361                              EM PRUDENTE DE MORAES   
30   280   ESCOLA MUNICIPAL GENERAL HUMBERTO DE SOUZA MELLO   
31   060               ESCOLA MUNICIPAL BARAO HOMEM DE MELO   
32   587                             EM ALMIRANTE TAMANDARE   
33   653                            EM LUCIA MIGUEL PEREIRA   
34   660                       E.M NOSSA SENHORA DA POMPEIA   
35   382                                    EM SANTOS ANJOS   
36   217                                         EM ALAGOAS   
37   378                           EM BRIGADEIRO FARIA LIMA   
38   429                                   EM FELIX PACHECO   
39   025                     ESCOLA MUNICIPAL GEORGE SUMMER   
40   660                       E.M NOSSA SENHORA DA POMPEIA   
41   373                       EM PROFESSOR AUGUSTO PAULINO   
42   624                               EM RIO GRANDE DO SUL   
43   157                                     E.M. SARMIENTO   
44   020                                          EM TAGORE   
45   117                        CIEP CHANCELER WILLY BRANDT   
46   502                               CIEP PATRICE LUMUMBA   
47   351  CENTRO INTEGRADO DE EDUCACAO PUBLICA VINICIUS ...   
48   135                                EM DOMINGOS BEBIANO   
49   388                                  EM CARDEAL CAMARA   
50   440                                 EM JOAQUIM RIBEIRO   
51   259                                      EM PERNAMBUCO   
52   422                 ESCOLA MUNICIPAL ALBINO SOUZA CRUZ   
53   041                                   EM ANDRADE NEVES   
54   537                                   EM ARY QUINTELLA   
55   230                                          EM BERLIM   
56   379                                CIEP NELSON MANDELA   
57   089                                   EM MARIO CLAUDIO   
58   146                            EM BERNARDO VASCONCELOS   
59   566                                     EM BRANT HORTA   
60   388                                  EM CARDEAL CAMARA   
61   498                                           EM CHILE   
62   663                          EM 

Identificar registros duplicados em `escolas`.

In [ ]:
duplicates_mask = escolas3.duplicated(subset='id',keep=False)
duplicates_df = escolas3[duplicates_mask]
duplicates_df.sort_values('id')

Eliminar registros duplicados de `escolas`.
Os registros de índices 117, 112 e 15 não foram eliminados por conter ligeiras diferenças com suas contrapartes, como a falta de uma vírgula, a presença de um caracter especial ou um erro de digitação. Dessa forma, sua eliminação foi manual.

* 117	206	EM HONDURAS - sem vírgula na latitude e longitude
* 112	379	CIEP NELSON MANDELA - º no endereço
* 15	413	EM CÓCIO BARCELOS - latitude incompleta

In [8]:
escolas4 = escolas3.drop_duplicates()
escolas4 = escolas4.drop([117,112,15]) # elimina escolas duplicadas, mas com ligeira diferença no registro
escolas4

id                                     escolas_postos  \
0    178        CENTRO INTEGRADO DE EDUCACAO PUBLICA HENFIL   
1    634                         EM ALICE DO AMARAL PEIXOTO   
2    483                                 EM CELESTINO SILVA   
3    476                  ESCOLA MUNICIPAL FLORIANO PEIXOTO   
4    132                                  EM PEREIRA PASSOS   
5    017                       E.M PRESIDENTE JOSE LINHARES   
6    055                                  EM SANTA CATARINA   
7    490                                      E.M TIA CIATA   
8    600                                         EM URUGUAI   
9    089                                   EM MARIO CLAUDIO   
10   387                     CIEP PRESIDENTE AGOSTINHO NETO   
11   331                                CIEP TANCREDO NEVES   
12   450                                   EM ALBERTO BARTH   
13   558                                     EM CICERO PENA   
14   413                                  EM COCIO BARCELOS   
16   540                              EM HENRIQUE DODSWORTH   
17   309                                          EM MEXICO   
18   268                      ESCOLA MUNICIPAL MINAS GERAIS   
19   523                     EM PRESIDENTE DA COSTA E SILVA   
21   035                                  EM SENADOR CORREA   
22   011                                 CIEP SAMUEL WEINER   
23   226                                 EM AFRANIO PEIXOTO   
24   638                                  E.M AZEVEDO SODRE   
26   126                                 EM DUQUE DE CAXIAS   
27   180                           ESCOLA MUNICIPAL EQUADOR   
28   163                                 EM LAUDIMIA TROTTA   
29   361                              EM PRUDENTE DE MORAES   
30   280   ESCOLA MUNICIPAL GENERAL HUMBERTO DE SOUZA MELLO   
31   060               ESCOLA MUNICIPAL BARAO HOMEM DE MELO   
32   587                             EM ALMIRANTE TAMANDARE   
33   653                            EM LUCIA MIGUEL PEREIRA   
34   660                       E.M NOSSA SENHORA DA POMPEIA   
35   382                                    EM SANTOS ANJOS   
36   217                                         EM ALAGOAS   
37   378                           EM BRIGADEIRO FARIA LIMA   
38   429                                   EM FELIX PACHECO   
39   025                     ESCOLA MUNICIPAL GEORGE SUMMER   
41   373                       EM PROFESSOR AUGUSTO PAULINO   
42   624                               EM RIO GRANDE DO SUL   
43   157                                     E.M. SARMIENTO   
44   020                                          EM TAGORE   
45   117                        CIEP CHANCELER WILLY BRANDT   
46   502                               CIEP PATRICE LUMUMBA   
47   351  CENTRO INTEGRADO DE EDUCACAO PUBLICA VINICIUS ...   
48   135                                EM DOMINGOS BEBIANO   
49   388                                  EM CARDEAL CAMARA   
50   440                                 EM JOAQUIM RIBEIRO   
51   259                                      EM PERNAMBUCO   
52   422                 ESCOLA MUNICIPAL ALBINO SOUZA CRUZ   
53   041                                   EM ANDRADE NEVES   
54   537                                   EM ARY QUINTELLA   
55   230                                          EM BERLIM   
56   379                                CIEP NELSON MANDELA   
58   146                            EM BERNARDO VASCONCELOS   
59   566                                     EM BRANT HORTA   
61   498                                           EM CHILE   
62   663                          EM CIENTISTA MARIO KROEFF   
63   121                                 EM DILERMANDO CRUZ   
64   161                                   E.M JOAO DE DEUS   
65   263                         EM PADRE MANUEL DE NOBREGA   
66   118   CENTRO INTEGRADO DE EDUCACAO PUBLICA ELIS REGINA   
67   140                                     EM PEDRO LESSA   
68   260                       EM PRO

Arredondar os campos 'latitude' e 'longitude' para cinco casas decimais.

In [9]:
escolas5 = escolas4
escolas5['latitude'] = escolas5['latitude'].str.replace(',', '.').astype(float).round(5)
escolas5['longitude'] = escolas5['longitude'].str.replace(',', '.').astype(float).round(5)

escolas5

id                                     escolas_postos  \
0    178        CENTRO INTEGRADO DE EDUCACAO PUBLICA HENFIL   
1    634                         EM ALICE DO AMARAL PEIXOTO   
2    483                                 EM CELESTINO SILVA   
3    476                  ESCOLA MUNICIPAL FLORIANO PEIXOTO   
4    132                                  EM PEREIRA PASSOS   
5    017                       E.M PRESIDENTE JOSE LINHARES   
6    055                                  EM SANTA CATARINA   
7    490                                      E.M TIA CIATA   
8    600                                         EM URUGUAI   
9    089                                   EM MARIO CLAUDIO   
10   387                     CIEP PRESIDENTE AGOSTINHO NETO   
11   331                                CIEP TANCREDO NEVES   
12   450                                   EM ALBERTO BARTH   
13   558                                     EM CICERO PENA   
14   413                                  EM COCIO BARCELOS   
16   540                              EM HENRIQUE DODSWORTH   
17   309                                          EM MEXICO   
18   268                      ESCOLA MUNICIPAL MINAS GERAIS   
19   523                     EM PRESIDENTE DA COSTA E SILVA   
21   035                                  EM SENADOR CORREA   
22   011                                 CIEP SAMUEL WEINER   
23   226                                 EM AFRANIO PEIXOTO   
24   638                                  E.M AZEVEDO SODRE   
26   126                                 EM DUQUE DE CAXIAS   
27   180                           ESCOLA MUNICIPAL EQUADOR   
28   163                                 EM LAUDIMIA TROTTA   
29   361                              EM PRUDENTE DE MORAES   
30   280   ESCOLA MUNICIPAL GENERAL HUMBERTO DE SOUZA MELLO   
31   060               ESCOLA MUNICIPAL BARAO HOMEM DE MELO   
32   587                             EM ALMIRANTE TAMANDARE   
33   653                            EM LUCIA MIGUEL PEREIRA   
34   660                       E.M NOSSA SENHORA DA POMPEIA   
35   382                                    EM SANTOS ANJOS   
36   217                                         EM ALAGOAS   
37   378                           EM BRIGADEIRO FARIA LIMA   
38   429                                   EM FELIX PACHECO   
39   025                     ESCOLA MUNICIPAL GEORGE SUMMER   
41   373                       EM PROFESSOR AUGUSTO PAULINO   
42   624                               EM RIO GRANDE DO SUL   
43   157                                     E.M. SARMIENTO   
44   020                                          EM TAGORE   
45   117                        CIEP CHANCELER WILLY BRANDT   
46   502                               CIEP PATRICE LUMUMBA   
47   351  CENTRO INTEGRADO DE EDUCACAO PUBLICA VINICIUS ...   
48   135                                EM DOMINGOS BEBIANO   
49   388                                  EM CARDEAL CAMARA   
50   440                                 EM JOAQUIM RIBEIRO   
51   259                                      EM PERNAMBUCO   
52   422                 ESCOLA MUNICIPAL ALBINO SOUZA CRUZ   
53   041                                   EM ANDRADE NEVES   
54   537                                   EM ARY QUINTELLA   
55   230                                          EM BERLIM   
56   379                                CIEP NELSON MANDELA   
58   146                            EM BERNARDO VASCONCELOS   
59   566                                     EM BRANT HORTA   
61   498                                           EM CHILE   
62   663                          EM CIENTISTA MARIO KROEFF   
63   121                                 EM DILERMANDO CRUZ   
64   161                                   E.M JOAO DE DEUS   
65   263                         EM PADRE MANUEL DE NOBREGA   
66   118   CENTRO INTEGRADO DE EDUCACAO PUBLICA ELIS REGINA   
67   140                                     EM PEDRO LESSA   
68   260                       EM PRO

Criar coluna `tipo_escola`

In [10]:
tipo_escola = {
'CENTRO INTEGRADO DE EDUCACAO PUBLICA':'CIEP', 
'E.M ':'EM ',
'E.M. ':'EM ',
'EM. ':'EM ', 
'ESCOLA MUNICIPAL ':'EM '
}

escolas6 = escolas5
for key, value in tipo_escola.items():
    escolas6['nome_escola'] = escolas6['nome_escola'].str.replace(fr'(?i)\b{key}\b', value, regex=True)

escolas6

id                               escolas_postos  \
0    178                                  CIEP HENFIL   
1    634                   EM ALICE DO AMARAL PEIXOTO   
2    483                           EM CELESTINO SILVA   
3    476                          EM FLORIANO PEIXOTO   
4    132                            EM PEREIRA PASSOS   
5    017                  EM PRESIDENTE JOSE LINHARES   
6    055                            EM SANTA CATARINA   
7    490                                 EM TIA CIATA   
8    600                                   EM URUGUAI   
9    089                             EM MARIO CLAUDIO   
10   387               CIEP PRESIDENTE AGOSTINHO NETO   
11   331                          CIEP TANCREDO NEVES   
12   450                             EM ALBERTO BARTH   
13   558                               EM CICERO PENA   
14   413                            EM COCIO BARCELOS   
16   540                        EM HENRIQUE DODSWORTH   
17   309                                    EM MEXICO   
18   268                              EM MINAS GERAIS   
19   523               EM PRESIDENTE DA COSTA E SILVA   
21   035                            EM SENADOR CORREA   
22   011                           CIEP SAMUEL WEINER   
23   226                           EM AFRANIO PEIXOTO   
24   638                             EM AZEVEDO SODRE   
26   126                           EM DUQUE DE CAXIAS   
27   180                                   EM EQUADOR   
28   163                           EM LAUDIMIA TROTTA   
29   361                        EM PRUDENTE DE MORAES   
30   280           EM GENERAL HUMBERTO DE SOUZA MELLO   
31   060                       EM BARAO HOMEM DE MELO   
32   587                       EM ALMIRANTE TAMANDARE   
33   653                      EM LUCIA MIGUEL PEREIRA   
34   660                  EM NOSSA SENHORA DA POMPEIA   
35   382                              EM SANTOS ANJOS   
36   217                                   EM ALAGOAS   
37   378                     EM BRIGADEIRO FARIA LIMA   
38   429                             EM FELIX PACHECO   
39   025                             EM GEORGE SUMMER   
41   373                 EM PROFESSOR AUGUSTO PAULINO   
42   624                         EM RIO GRANDE DO SUL   
43   157                                 EM SARMIENTO   
44   020                                    EM TAGORE   
45   117                  CIEP CHANCELER WILLY BRANDT   
46   502                         CIEP PATRICE LUMUMBA   
47   351                      CIEP VINICIUS DE MORAES   
48   135                          EM DOMINGOS BEBIANO   
49   388                            EM CARDEAL CAMARA   
50   440                           EM JOAQUIM RIBEIRO   
51   259                                EM PERNAMBUCO   
52   422                         EM ALBINO SOUZA CRUZ   
53   041                             EM ANDRADE NEVES   
54   537                             EM ARY QUINTELLA   
55   230                                    EM BERLIM   
56   379                          CIEP NELSON MANDELA   
58   146                      EM BERNARDO VASCONCELOS   
59   566                               EM BRANT HORTA   
61   498                                     EM CHILE   
62   663                    EM CIENTISTA MARIO KROEFF   
63   121                           EM DILERMANDO CRUZ   
64   161                              EM JOAO DE DEUS   
65   263                   EM PADRE MANUEL DE NOBREGA   
66   118                             CIEP ELIS REGINA   
67   140                               EM PEDRO LESSA   
68   260                 EM PROFESSOR JOSUE DE CASTRO   
69   069                         EM CARDEAL ARCOVERDE   
70   090                          EM THOMAS JEFFERSON   
71   274                            EM ABEILARD FEIJO   
72   676                           EM ANISIO TEIXEIRA   
73   390                     EM ANDREA FONTES PEIXOTO   
74   608                     EM GRANDJEAN DE MONTIGNY   
75   589        

Criar coluna `tipo_escola` com valores extraídos de `escolas_postos` (EM|CIEP|COLEGIO MUNICIPAL).

In [13]:
escolas7 = escolas6
escolas7['tipo_escola'] = escolas7['nome_escola'].str.extract(r'(EM|CIEP|COLEGIO MUNICIPAL)', expand=False)
escolas7['nome_escola'] = escolas7['nome_escola'].str.replace(r'(EM|CIEP|COLEGIO MUNICIPAL)', '', regex=True).str.strip()
cols = ['id','nome_escola','tipo_escola','logradouro','bairro','latitude','longitude']
escolas7 = escolas7[cols]
escolas7

id        tipo_escola                           escolas_postos  \
0    178               CIEP                                   HENFIL   
1    634                 EM                  ALICE DO AMARAL PEIXOTO   
2    483                 EM                          CELESTINO SILVA   
3    476                 EM                         FLORIANO PEIXOTO   
4    132                 EM                           PEREIRA PASSOS   
5    017                 EM                 PRESIDENTE JOSE LINHARES   
6    055                 EM                           SANTA CATARINA   
7    490                 EM                                TIA CIATA   
8    600                 EM                                  URUGUAI   
9    089                 EM                            MARIO CLAUDIO   
10   387               CIEP                PRESIDENTE AGOSTINHO NETO   
11   331               CIEP                           TANCREDO NEVES   
12   450                 EM                            ALBERTO BARTH   
13   558                 EM                              CICERO PENA   
14   413                 EM                           COCIO BARCELOS   
16   540                 EM                       HENRIQUE DODSWORTH   
17   309                 EM                                   MEXICO   
18   268                 EM                             MINAS GERAIS   
19   523                 EM              PRESIDENTE DA COSTA E SILVA   
21   035                 EM                           SENADOR CORREA   
22   011               CIEP                            SAMUEL WEINER   
23   226                 EM                          AFRANIO PEIXOTO   
24   638                 EM                            AZEVEDO SODRE   
26   126                 EM                          DUQUE DE CAXIAS   
27   180                 EM                                  EQUADOR   
28   163                 EM                          LAUDIMIA TROTTA   
29   361                 EM                       PRUDENTE DE MORAES   
30   280                 EM          GENERAL HUMBERTO DE SOUZA MELLO   
31   060                 EM                        BARAO HOM DE MELO   
32   587                 EM                      ALMIRANTE TAMANDARE   
33   653                 EM                     LUCIA MIGUEL PEREIRA   
34   660                 EM                 NOSSA SENHORA DA POMPEIA   
35   382                 EM                             SANTOS ANJOS   
36   217                 EM                                  ALAGOAS   
37   378                 EM                    BRIGADEIRO FARIA LIMA   
38   429                 EM                            FELIX PACHECO   
39   025                 EM                            GEORGE SUMMER   
41   373                 EM                PROFESSOR AUGUSTO PAULINO   
42   624                 EM                        RIO GRANDE DO SUL   
43   157                 EM                                SARMIENTO   
44   020                 EM                                   TAGORE   
45   117               CIEP                   CHANCELER WILLY BRANDT   
46   502               CIEP                          PATRICE LUMUMBA   
47   351               CIEP                       VINICIUS DE MORAES   
48   135                 EM                         DOMINGOS BEBIANO   
49   388                 EM                           CARDEAL CAMARA   
50   440                 EM                          JOAQUIM RIBEIRO   
51   259                 EM                               PERNAMBUCO   
52   422                 EM                        ALBINO SOUZA CRUZ   
53   041                 EM                            ANDRADE NEVES   
54   537                 EM                            ARY QUINTELLA   
55   230                 EM                                   BERLIM   
56   379               CIEP                           NELSON MANDELA   
58   146                 EM                     BERNARDO VASCONCELOS   
59   566                 EM                   

In [ ]:
Corrige as abreviações nos endereços.

In [14]:
tipo_endereco = {
    'AV ':'AVENIDA', 
    'AV. ':'AVENIDA ',
    'ESTR. ':'ESTRADA ',
    'PCA. ': 'PRACA ',
    'R. ':'RUA ',
    S/No
    S/N.o
    
}

escolas8 = escolas7
for key, value in tipo_endereco.items():
    escolas8['endereco'] = escolas8['endereco'].str.replace(fr'(?i)\b{key}\b', value, regex=True)

escolas8

id        tipo_escola                           escolas_postos  \
0    178               CIEP                                   HENFIL   
1    634                 EM                  ALICE DO AMARAL PEIXOTO   
2    483                 EM                          CELESTINO SILVA   
3    476                 EM                         FLORIANO PEIXOTO   
4    132                 EM                           PEREIRA PASSOS   
5    017                 EM                 PRESIDENTE JOSE LINHARES   
6    055                 EM                           SANTA CATARINA   
7    490                 EM                                TIA CIATA   
8    600                 EM                                  URUGUAI   
9    089                 EM                            MARIO CLAUDIO   
10   387               CIEP                PRESIDENTE AGOSTINHO NETO   
11   331               CIEP                           TANCREDO NEVES   
12   450                 EM                            ALBERTO BARTH   
13   558                 EM                              CICERO PENA   
14   413                 EM                           COCIO BARCELOS   
16   540                 EM                       HENRIQUE DODSWORTH   
17   309                 EM                                   MEXICO   
18   268                 EM                             MINAS GERAIS   
19   523                 EM              PRESIDENTE DA COSTA E SILVA   
21   035                 EM                           SENADOR CORREA   
22   011               CIEP                            SAMUEL WEINER   
23   226                 EM                          AFRANIO PEIXOTO   
24   638                 EM                            AZEVEDO SODRE   
26   126                 EM                          DUQUE DE CAXIAS   
27   180                 EM                                  EQUADOR   
28   163                 EM                          LAUDIMIA TROTTA   
29   361                 EM                       PRUDENTE DE MORAES   
30   280                 EM          GENERAL HUMBERTO DE SOUZA MELLO   
31   060                 EM                        BARAO HOM DE MELO   
32   587                 EM                      ALMIRANTE TAMANDARE   
33   653                 EM                     LUCIA MIGUEL PEREIRA   
34   660                 EM                 NOSSA SENHORA DA POMPEIA   
35   382                 EM                             SANTOS ANJOS   
36   217                 EM                                  ALAGOAS   
37   378                 EM                    BRIGADEIRO FARIA LIMA   
38   429                 EM                            FELIX PACHECO   
39   025                 EM                            GEORGE SUMMER   
41   373                 EM                PROFESSOR AUGUSTO PAULINO   
42   624                 EM                        RIO GRANDE DO SUL   
43   157                 EM                                SARMIENTO   
44   020                 EM                                   TAGORE   
45   117               CIEP                   CHANCELER WILLY BRANDT   
46   502               CIEP                          PATRICE LUMUMBA   
47   351               CIEP                       VINICIUS DE MORAES   
48   135                 EM                         DOMINGOS BEBIANO   
49   388                 EM                           CARDEAL CAMARA   
50   440                 EM                          JOAQUIM RIBEIRO   
51   259                 EM                               PERNAMBUCO   
52   422                 EM                        ALBINO SOUZA CRUZ   
53   041                 EM                            ANDRADE NEVES   
54   537                 EM                            ARY QUINTELLA   
55   230                 EM                                   BERLIM   
56   379               CIEP                           NELSON MANDELA   
58   146                 EM                     BERNARDO VASCONCELOS   
59   566                 EM                   

Carregar o dataframe `material_didatico` do arquivo csv.

In [ ]:
material_didatico = pd.read_csv('data/material_didatico.csv')

column_mapping = {
    'id': 'id',
    'Quantidade': 'qt_material_didatico'
}

material_didatico.rename(columns=column_mapping,inplace=True)
material_didatico

In [ ]:
material_didatico['id'] = material_didatico['id'].astype(str).str.zfill(3)
material_didatico

In [ ]:
duplicates_mask = material_didatico.duplicated(subset='id',keep=False)
duplicates_df = material_didatico[duplicates_mask]
duplicates_df.sort_values('id')

In [ ]:
subprefeituras = pd.read_csv('data/subprefeituras.csv')

column_mapping = {
    'nome': 'bairro',
    'subprefeitura': 'subprefeitura'
}

subprefeituras.rename(columns=column_mapping,inplace=True)
subprefeituras['bairro'] = subprefeituras['bairro'].str.title()
subprefeituras

In [ ]:
duplicates_mask = subprefeituras.duplicated(subset='bairro',keep=False)
duplicates_df = subprefeituras[duplicates_mask]
duplicates_df.sort_values('bairro')

In [ ]:
df_temp = pd.merge(escolas_unique,material_didatico,on='id', how='left')

df_temp

In [ ]:
df_final = pd.merge(df_temp,subprefeituras,on='bairro',how='left')

df_final

In [ ]:
df_final.describe()

In [ ]:
df_final.info()

In [ ]:
df_final[df_final['subprefeitura'].isnull()]

In [ ]:
qt_mask = pd.to_numeric(df_final['qt_material_didatico'], errors='coerce').isna()
not_numeric = df_final[qt_mask]
not_numeric

In [ ]:
subprefeituras_lst = list(subprefeituras['bairro'].unique())
subprefeituras_lst.sort()
subprefeituras_lst